component/componenterror/permanent.go

```diff
@@ -0,0 +1,48 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package componenterror // import "go.opentelemetry.io/collector/component/componenterror"
```

Comment:
Is there a good, common place we can put them for use in the `component` and `consumer` packages? If so, we should try to consolidate them.

Reply:
What does permanent error mean here? I think the discussion was to have this more or less like an error with a "code/state" rather than wrapping error.